net/retriever/retriever_test.go

```diff
@@ -0,0 +1,39 @@
+package net
+
+import (
+	"testing"
+
+	"github.com/OpenBazaar/openbazaar-go/pb"
+)
+
+// TestEnsureNoOmissionsInMessageProcessingOrder ensures that
+// new message types are either included in MessageProcessingOrder
+// or added to the BlackList, indicating it should not be added.
+func TestEnsureNoOmissionsInMessageProcessingOrder(t *testing.T) {
+	messages := make(map[string]int32, len(pb.Message_MessageType_value))
+	for k, v := range pb.Message_MessageType_value {
+		messages[k] = v
+	}
+
+	// Add deliberate omissions to this list
+	BlackList := map[pb.Message_MessageType]struct{}{
+		pb.Message_PING:          struct{}{},
+		pb.Message_OFFLINE_RELAY: struct{}{},
+		pb.Message_STORE:         struct{}{},
+		pb.Message_BLOCK:         struct{}{},
+		pb.Message_ERROR:         struct{}{},
+	}
+
+	// Inclusion check
+	for _, msgType := range MessageProcessingOrder {
+		delete(messages, msgType.String())
+	}
+	// BlackList check
+	for msgType := range BlackList {
+		delete(messages, msgType.String())
+	}
+
+	if l := len(messages); l > 0 {
+		t.Errorf("Found %d omission(s): %v", l, messages)
```

Comment:
I like for unit test errors to describe the expected outcome:
`"found %d unexpected message types which are not considered in MessageProcessOrder: %v"`

(Note the error message is not capitalized... this is also idiomatic in go error messages generally.)